### ISSUE TITLE
Run heuristic data mix sweep

### ISSUE THREAD

**Issue by eric-czech** [Open-Athena/bolinas-dna#135]
refs: https://github.com/Open-Athena/bolinas-dna/issues/109
refs: https://openathena.slack.com/archives/C0A7JN194MT/p1777035229742039

WIP

<details><summary>Commands</summary>

```
for mix in uniform cds_only upstream_only downstream_only; do
  DATE=$(date +%Y%m%d-%H%M%S)
  uv run iris --cluster=marin job run --zone us-east5-a --user eczech --no-wait \
    --job-name "dna-bolinas-mix-v0.1-${mix}-${DATE}" \
    -e WANDB_API_KEY "$WANDB_API_KEY" \
    -e SWEEP_MIX_NAMES "${mix}" \
    -- python -m experiments.dna.exp135_bolinas_mixing_sweep
  sleep 1
done
```

</details>

**Comment by eric-czech** [comment]
## Results

WIP

## Learning trajectories

Trajectories for the model lineages, i.e. series of connected pretraining runs, built on the expanded zoonomia dataset (w/ enhancers):

<p align="center">
<img width="768" alt="Image" src="https://github.com/user-attachments/assets/9a098105-5a63-4af1-844c-11b95984aebc" />
</p>

This contrasts with many of the earlier, less performant models built solely on metazoan sequence data (i.e. animals).

## Leaderboard scores

A snapshot of select model scores from https://openathena.ai/marin-dna/leaderboards/mendelian (2026-06-25):

<img width="768" alt="Image" src="https://github.com/user-attachments/assets/b555df38-e4b7-450e-82fc-5f89f3083871" />

The `m5.1` model is the best one from this experiment.

## Region transfer

Token-matched loss and VEP scores from earlier proportionally mixed models (see https://github.com/Open-Athena/marin-dna/issues/109) and the uniform mixture models used in this line of experiments:

<img width="896" alt="Image" src="https://github.com/user-attachments/assets/06f7a5b8-3c54-4fc4-913c-182e38eb4197" />

**Note**: `pooled` = proportional mix, `unpooled` = uniform mix

This is saying:

1. 3' UTR VEP performance actually improves some if the `i`th downstream token is preceded by tokens from other regions 
    - By tokens from other regions, I mean that the proportional mix models are seeing 7 CDS and 2 upstream tokens for every 1 downstream token rather than 1:1:1 for the uniform mix models
    - This suggests some kind of production transfer is possible for them from other genomic regions
2. Upstream tasks (5' UTR and promoter / `tss_proximal`) work the opposite way and to a far greater extent
    - I.e. seeing as few non-upstream tokens as possible appears to maximize their performance
    - This suggests they are most sensitive to interference from other genomic regions
3. Other tasks like splice site and synonymous variants maybe show a little positive synergy like 3' UTRs
    - None of that outweighs the huge drop in upstream performance resulting from proportional mixing
    - I.e. this is a decomposition adding a little more color to why uniform models perform much better in aggregate 

## Lineage scores

This whole experiment involved a lot of heuristic experimentation based on vibes and educated guesses.  Here is a tree helping to demonstrate the relationship between different model lineages and to some extent, what branches from certain pretraining checkpoints were most productive:

<img width="896" alt="Image" src="https://github.com/user-attachments/assets/1d6711ea-6eff-4e31-a184-4a77ce92f88d" />

## Cooldown effects

One question I had throughout this sweep is whether or not it is ever worth using a decaying LR.  That clearly improves on loss nats in the way we'd expect, but the effect on VEP was less clear to me.  I ran an analysis on the `m1.x` and `m3.x` lineages of this script to gather better evidence one way or the other.  Each of these two lineages consists of 3 overlapping runs where each is continued from the previous prior to cooldown.  This provides a convenient counterfactual scenario where I can compare changes in nats or VEP AUPRC after cooldown to what would have happened if LR was never decayed and training continued over the same number of cooldown tokens.  I'm looking at these here as a `speedup` defined as how many more times the post cooldown metric improved over training without cooldown:

  | metric                    | group   | Δ cooldown | Δ continued | speedup |
  |---------------------------|---------|:----------:|:-----------:|:-------:|
  | Validation loss           | m1.x    |   −0.042   |   −0.011    |  3.8×   |
  | Validation loss           | m3.x    |   −0.043   |   −0.009    |  4.5×   |
  | Validation loss           | Overall |   −0.042   |   −0.010    |  4.1×   |
  | VEP macro avg (8 subsets) | m1.x    |   +0.0406  |   +0.0173   |  2.4×   |
  | VEP macro avg (8 subsets) | m3.x    |   +0.0326  |   +0.0119   |  2.7×   |
  | VEP macro avg (8 subsets) | Overall |   +0.0366  |   +0.0146   |  2.5×   |

<details><summary>Details by lineage</summary>

# Cooldown effects (m1 & m3 continuation chains)

A natural counterfactual from the two zoonomia pre-cooldown chains (`m1→m1.1→m1.2→m1.3`, `m3→m3.1→m3.2→m3.3`). At each fork a child resumes from the parent **before** the parent's learning-rate cooldown and keeps training at peak LR, while the parent finishes its own cooldown on a path the child never took. This lets us compare, at matched cumulative-token levels:

- **pre-cooldown** — the parent's eval at the fork (shared warm-start checkpoint, peak LR).
- **post-cooldown** — the parent's final eval after it cooled down.
- **continued (peak LR)** — the child's eval at the *same* cumulative tokens as the parent's cooled end, reached by staying at peak LR.

**Deltas** (relative to pre-cooldown): `Δ cooldown = post − pre` (effect of cooling down) and `Δ continued = continued − pre` (effect of spending the same extra tokens at peak LR instead). For `eval/loss` lower is better, so an improvement is **negative**; for VEP AUPRC higher is better, so an improvement is **positive**.

**speedup** `= Δ cooldown / Δ continued`: how many times larger the gain from cooling down is than the gain from spending the same extra tokens at peak LR. A value of e.g. 3× means cooldown achieved 3× the metric change that continued peak-LR training did over the same token budget. (Both deltas share a sign per metric, so the ratio is positive.)

> **Note on the terminal forks** (`m1.2→m1.3`, `m3.2→m3.3`): the child is the terminal leaf, which begins its own cooldown before reaching the parent's final token level, so its *continued (peak LR)* point is ~29% into the leaf's cooldown rather than pure peak LR. Those rows are flagged and excluded from the clean-fork averages below. The pre→post cooldown delta is unaffected.

Full per-metric data (all 8 VEP subsets) is in `cooldown_effects.csv`.

## Validation loss (`eval/loss`)

| fork | tokens (B) | pre-cooldown | post-cooldown | continued (peak LR) | Δ cooldown | Δ continued | speedup | note |
|---|---|---|---|---|---|---|---|---|
| m1->m1.1 | 50→62 | 1.169 | 1.135 | 1.158 | -0.034 | -0.010 | 3.2× |  |
| m1.1->m1.2 | 87→112 | 1.144 | 1.094 | 1.132 | -0.050 | -0.012 | 4.3× |  |
| m1.2->m1.3 | 112→149 | 1.132 | 1.072 | 1.105 | -0.060 | -0.027 | 2.3× | continued pt 29% into leaf cooldown |
| m3->m3.1 | 50→62 | 1.166 | 1.131 | 1.158 | -0.035 | -0.009 | 4.0× |  |
| m3.1->m3.2 | 87→112 | 1.140 | 1.090 | 1.130 | -0.050 | -0.010 | 5.0× |  |
| m3.2->m3.3 | 112→149 | 1.130 | 1.069 | 1.103 | -0.061 | -0.027 | 2.3× | continued pt 28% into leaf cooldown |
| **m1.x mean** | — | — | — | — | **-0.042** | **-0.011** | **3.8×** | mean of 2 clean forks |
| **m3.x mean** | — | — | — | — | **-0.043** | **-0.009** | **4.5×** | mean of 2 clean forks |

Mean over the 4 clean forks: Δ cooldown = -0.042, Δ continued = -0.010, speedup = 4.1×.

## VEP macro average (mean of 8 subsets)

| fork | tokens (B) | pre-cooldown | post-cooldown | continued (peak LR) | Δ cooldown | Δ continued | speedup | note |
|---|---|---|---|---|---|---|---|---|
| m1->m1.1 | 50→62 | 0.3489 | 0.3833 | 0.3672 | +0.0344 | +0.0182 | 1.9× |  |
| m1.1->m1.2 | 87→112 | 0.3759 | 0.4227 | 0.3922 | +0.0468 | +0.0163 | 2.9× |  |
| m1.2->m1.3 | 112→149 | 0.3922 | 0.4368 | 0.4189 | +0.0446 | +0.0267 | 1.7× | continued pt 29% into leaf cooldown |
| m3->m3.1 | 50→62 | 0.3690 | 0.3972 | 0.3756 | +0.0282 | +0.0066 | 4.3× |  |
| m3.1->m3.2 | 87→112 | 0.3921 | 0.4291 | 0.4092 | +0.0370 | +0.0171 | 2.2× |  |
| m3.2->m3.3 | 112→149 | 0.4092 | 0.4280 | 0.4292 | +0.0188 | +0.0200 | 0.9× | continued pt 28% into leaf cooldown |
| **m1.x mean** | — | — | — | — | **+0.0406** | **+0.0173** | **2.4×** | mean of 2 clean forks |
| **m3.x mean** | — | — | — | — | **+0.0326** | **+0.0119** | **2.7×** | mean of 2 clean forks |

Mean over the 4 clean forks: Δ cooldown = +0.0366, Δ continued = +0.0146, speedup = 2.5×.



</details>

Somewhat to my surprise, cooldown actually seems to help quite a bit on VEP.  The relative improvement is maybe a little less drastic than it is for loss, but this suggests that continuing to spend ~20% of the FLOPS for each run on it is justified.  The details in dropdown make it more clear that these averages are over values that all have the same sign and consistent magnitudes.
